File: src/main/java/org/ximo/finkinaction/java/datastream/timewindow/StreamTimeCharacteristicApp.java
package org.ximo.finkinaction.java.datastream.timewindow;

import org.apache.flink.streaming.api.TimeCharacteristic;
import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;

/**
 * @author xikl
 * @date 2019/9/10
 */
public class StreamTimeCharacteristicApp {
    public static void main(String[] args) {

        StreamExecutionEnvironment environment = StreamExecutionEnvironment.getExecutionEnvironment();

        environment.setStreamTimeCharacteristic(TimeCharacteristic.EventTime);
        // alternatively:
        // env.setStreamTimeCharacteristic(TimeCharacteristic.IngestionTime);
        // env.setStreamTimeCharacteristic(TimeCharacteristic.EventTime);


    }

}

File: src/main/scala/org/ximo/finkinaction/scala/datastream/timewindow/StreamTimeCharacteristicApp.scala
package org.ximo.finkinaction.scala.datastream.timewindow

import org.apache.flink.streaming.api.TimeCharacteristic
import org.apache.flink.streaming.api.scala.StreamExecutionEnvironment

/**
  *
  *
  * @author xikl
  * @date 2019/9/10
  */
object StreamTimeCharacteristicApp extends App {

  val env = StreamExecutionEnvironment.getExecutionEnvironment

  env.setStreamTimeCharacteristic(TimeCharacteristic.EventTime)



}
